### ISSUE TITLE
Fix dockerhub push on release

### ISSUE THREAD

**Issue by i-norden** [vulcanize/ipld-eth-db#35]
This is a problem across multiple repos. We shouldn't need to merge something into `main` in order for it to be published to dockerhub when a release is cut.

**Comment by abdulrabbani00** [comment]
@ashwinphatak - Would putting together a manual CI/CD pipeline be better? Similar to the demo I gave for `ipld-ethcl-indexer`. This is what it might look like:

1. User provides inputs:
  a. branch
  b. tag name for docker.
2. Docker creates and tags the image.
3. The image gets published to the registry.

This way users can hypothetically publish images on working branches and don't have to rely on the image getting published to main.

**Comment by ashwinphatak** [comment]
The suggested approach provides a lot of flexibility, but we should consider if that flexibility is needed. The other option is to trigger docker image creation and publish when a new release is cut (for example, from the `sharding` branch).

**Comment by abdulrabbani00** [comment]
# Overview
We need to update the following repositories with this request:

# Repositories

## Critical Priority
Repositories that need to be updated immediately.

- [x] `ipld-eth-db` - @abdulrabbani00 - https://github.com/vulcanize/ipld-eth-db/pull/77

## Top Priority

Repositories that need to be updated very soon, and should be prioritized accordingly.

- [x] `vulcanize-geth` - @abdulrabbani00 
- [x] `ipld-eth-server` - @abdulrabbani00 - https://github.com/vulcanize/ipld-eth-server/pull/151 - Tests are failing even though no application changes have been maded.
- [x] ~~`ipfs-ethdb`~~ - @abdulrabbani00  - This repository does not contain a Dockerfile. If it actually needs to containerized, then we should create a task for it.
- [x] ~~`eth-ipfs-state-validator`~~ - @abdulrabbani00 - This repository does not contain a Dockerfile. If it actually needs to containerized, then we should create a task for it.
- [x] ~~`ipld-eth-db-validator`~~ - @abdulrabbani00 - This repository does not contain a Dockerfile. If it actually needs to containerized, then we should create a task for it.
- [x] ~~`ipld-eth-state-snapshot`~~ - @abdulrabbani00 - This repository does not contain a Dockerfile. If it actually needs to containerized, then we should create a task for it.
- [x] `eth-statediff-service` - @abdulrabbani00 

## Low Priority

Repositories that can be lazily updated.

 - [ ] Pass

**Comment by abdulrabbani00** [comment]
I've completed the first item. To achieve this I did the following:

1. I combined the build and push jobs into one workflow.
2. I made the `push` job `needs` the `build` job. This way they don't run in parallel.

* Run example: https://github.com/vulcanize/ipld-eth-db/runs/6422612226?check_suite_focus=true
* Code to reference: https://github.com/vulcanize/ipld-eth-db/blob/feature/docker-cicd/.github/workflows/publish.yaml
